src/main.rs

```diff
@@ -1,95 +0,0 @@
-extern crate sdl2;
-
-use crate::util::*;
-use gamestate::GameState;
-use images::Images;
-use renderer::Renderer;
-use sdl2::event::Event;
-use sdl2::image::InitFlag;
-use sdl2::keyboard::Keycode;
-
-const BOARD_LENGTH: i32 = 8;
-const TILE_SIZE: i32 = 96;
-
-mod images;
-mod renderer;
-mod util;
-
-mod gamestate;
-
-fn main() -> Result<(), String> {
-    let sdl_context = sdl2::init()?;
-    let video_subsystem = sdl_context.video()?;
-
-    let _image_context = sdl2::image::init(InitFlag::PNG);
-
-    let window_length = (BOARD_LENGTH * TILE_SIZE) as u32;
-
-    let window = video_subsystem
-        .window("chess", window_length, window_length)
-        .position_centered()
-        .opengl()
-        .build()
-        .map_err(|e| e.to_string())?;
-
-    let canvas = window
-        .into_canvas()
-        .software()
-        .build()
-        .map_err(|e| e.to_string())?;
-
-    let texture_creator = canvas.texture_creator();
-
-    let mut renderer = Renderer::new(canvas, Images::new(&texture_creator)?)?;
-    let mut gamestate = GameState::new();
-
-    let mut game_continue = true;
-
-    'mainloop: loop {
-        for event in sdl_context.event_pump()?.poll_iter() {
-            match event {
-                Event::Quit { .. }
-                | Event::KeyDown {
-                    keycode: Option::Some(Keycode::Escape),
-                    ..
-                } => break 'mainloop,
```

Comment:
why use a labelled loop over just return Ok(())?

Reply:
ah i just learned this from sdl2 rust tutorials thanks never thought about that  cause i dont do any work after i think its for if you wanna do stuff after the gameloop end